### PR TITLE
Revert changes from PR #1390 to see if Travis succeeds

### PR DESCRIPTION
**Pull request by tjguk**

### CODE REVIEW

File: tests/virtual_environment/test_virtual_environment.py [hunk 1]
```diff
@@ -43,11 +43,8 @@
 def venv_name():
     """Use a random venv name each time, at least partly to expose any
     hidden assumptions about the name of the venv directory
-
-    Force the name to have a space in it to expose possible fragilities that way
     """
-    hex = uuid.uuid1().hex
-    return hex[:2] + " " + hex[2:4]
```
Comment (carlosperate) [inline tests/virtual_environment/test_virtual_environment.py:50]: I had a quick look at this, and I think the issue is to do with the spaces on the virtualenv pip.

When trying to use the venv pip I get the following error in the terminal due to the spaces in the shebang:
```
$  ls -a /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/
.   activate      activate.fish     easy_install      pip   pip3.5  python3    python-config
..  activate.csh  activate_this.py  easy_install-3.5  pip3  python  python3.5  wheel


$ /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/pip --version
bash: /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/bin/pip: "/tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7: bad interpreter: No such file or directory
```
`pip`  file content (trying to scape the space with `\ ` doesn't make a differnece):
```python
#!"/tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/bin/python3"

# -*- coding: utf-8 -*-
import re
import sys

from pip._internal.cli.main import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

This works:

```
$ /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/python -m pip --version
pip 20.3.4 from /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/lib/python3.5/site-packages/pip (python 3.5)
```

**Comment by tjguk** [comment]
Thanks for the research, @carlosperate. So, amusingly, while I still can't reproduce the original Windows-based problem with username spaces, by adding a test we've highlighted a *nix problem?

**Comment by carlosperate** [comment]
Yeah, quite possibly, but it's more likely a virtualenv version requirement.
I'm still not 100% sure this is the problem, but it looks very promising.

If the issue is on the shebang itself breaking in *nix environments, this fix is likely in an updated version of virtualenv. I didn't read this entire thread, but there is probably something here:
https://github.com/pypa/virtualenv/issues/53

The version in CI is 15.1, the one I used in the VM was 15.0.1 (the one packaged by Ubuntu), I wouldn't be surprised if this was fixed in a later version. Unfortunately when trying to pip install it in the Mu venv (the one manually created to run the tests, not the one Mu creates), then it updates other packages that seem broken in Python 3.5.

For comparison, the pip file in my macOS mu venv has this type of shebang:
```
#!/bin/sh
'''exec' "/Users/microbit-carlos/Library/Application Support/mu/mu_venv-38-20210322-132556/bin/python" "$0" "$@"
' '''
# -*- coding: utf-8 -*-
```

I can invoke pip if I copy this style shebang (and pip code) in the Ubuntu virtual environment where pip threw the ` bad interpreter: No such file or directory` error:
```
$ /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/pip list
DEPRECATION: Python 3.5 reached the end of its life on September 13th, 2020. Please upgrade your Python as Python 3.5 is no longer maintained. pip 21.0 will drop support for Python 3.5 in January 2021. pip 21.0 will remove support for this functionality.
Package       Version
------------- -------
pip           20.3.4
pkg-resources 0.0.0
setuptools    51.2.0
wheel         0.36.2
```

**Comment by carlosperate** [comment]
I'll add this here to capture it somewhere. This is the issue I'm getting when virtualenv is pip installed in the venv instead of using the Ubuntu packaged version:
```
$ /home/carlos/workspace/mu/venv/bin/python3 -m virtualenv  -p /home/carlos/workspace/mu/venv/bin/python3 -q "/tmp/pytest-of-carlos/pytest-33/test_create_virtual_environmen0/f1 1c"
ImportError: cannot import name 'ContextManager'
```

Looks like something is importing the `ContextManager` typing which was only added in 3.5.4, and Ubuntu 16.04 is at 3.5.1 or 3.5.2:
- https://docs.python.org/3/library/typing.html#typing.ContextManager
- https://packages.ubuntu.com/xenial/python/python3
- https://packages.ubuntu.com/xenial/python/python3.5

**Comment by carlosperate** [comment]
Okay, I was able to fix the state of the packages in the system python site-packages and can confirm that specifying a minimum version of `virtualenv` fixes this issue: https://github.com/mu-editor/mu/pull/1400

**Pull request merged**